doc/contributing/writing-tests.md

```diff
@@ -141,6 +141,26 @@ request. Interesting things to notice:
 
 ## General recommendations
 
+### Usage of `node:test`
+
+It is recommended to use `node:test` in tests outside of testing the `node:test`
+module, if particular functionality that's tested is not a dependency of
+`node:test` module. This ensures that a bug in the test runner doesn't impact
+the outcome of the underlying dependencies' test results.
+
+These dependencies are:
+
+- `node:async_hooks`
```

Comment:
To be honest, `async_hooks` is probably the only thing I would include here (and I may update the test runner to migrate off of that in the future).

Reply:
Sounds good to me, but before removing the rest, what's your reasoning for keeping only `async_hooks` here?

Reply:
`async_hooks` actually changes how things work.

`child_process` and `fs` are so heavily depended on by other things that if they stop working we will definitely notice. The test runner also doesn't do anything "fancy" with them.

The only place the test runner uses a stream is for emitting events. If you were going to include that, you may as well include event emitter as well since it is part of streams.

The `vm` module is only used (directly) for evaluating snapshot files.

Reply:
Can you recommend changes to the text, please?